truffleruby/src/main/c/cext/ruby.c

```diff
@@ -379,15 +379,15 @@ int NUM2INT(VALUE value) {
 }
 
 unsigned int NUM2UINT(VALUE value) {
-  return (unsigned int) truffle_invoke_i(RUBY_CEXT, "NUM2UINT", value);
+  return (unsigned int) truffle_invoke_l(RUBY_CEXT, "NUM2LONG", value);
```

Comment:
NUM2UINT is effectively a truncation of NUM2LONG, and must handle conversion of numbers that don't fit into a signed Java int.

We should probably revisit these CextNodes anyway as ruby defines these to be either macros wrapping the rb_num2xxx variants, or inlined versions of them so the current CextNodes aren't complete in their behaviour. I don't think there are any specs for most of these though.